### ISSUE TITLE
Sending mail via university office365.com account fails

### ISSUE THREAD

**Issue by detrout** [flathub/org.kde.kontact#10]
Hi.

I was able to successfully send email via the older debian testing kmail 5.7.3 version via smtp.office365.com. But the newer flatpak kmail 5.9.1 can't send. Even with as close to the same settings as I can get. (Debian kmail is using TLS/port 25/LOGIN), I tried that with flatpak 5.9.1, as well and it also fails. (Though flatpak prefers STARTTLS/port 587/LOGIN)

org.kde.pim.ksmtp: Socket error: 2 "The remote host closed the connection"

Debian has a /usr/lib/x86_64-linux-gnu/sasl2/liblogin.so.2 library, the flatpak container doesn't. Should it? (Here's what I found /usr/lib/libsasl2 )

```
libanonymous.so        libdigestmd5.so.3      libplain.so.3.0.0
libanonymous.so.3      libdigestmd5.so.3.0.0  libsasldb.so
libanonymous.so.3.0.0  libotp.so              libsasldb.so.3
libcrammd5.so          libotp.so.3            libsasldb.so.3.0.0
libcrammd5.so.3        libotp.so.3.0.0        libscram.so
libcrammd5.so.3.0.0    libplain.so            libscram.so.3
libdigestmd5.so        libplain.so.3          libscram.so.3.0.0
```

**Comment by danvratil** [comment]
Hi, thanks for your vigorous testing :-) 

Sasl comes from org.freedesktop.Platform runtime, which is the low-level runtime that even our KDE Runtime is built on, so either we have to fix it in there, or I'll have to compile our own Sasl with LOGIN support.

**Comment by danvratil** [comment]
Thanks, Diane. Let me know if you need any help. 

And keep them coming - I can't promise I'll be able to fix all the issues, but it's important that the Flatpak works as best as possible for everyone :)

**Comment by detrout** [comment]
I was able to send email via office365 with this patch added.

I didn't make a pull request yet because I'm not completely sure what version to use. 2.1.26 was my best guess for what was in the freedesktop 1.6 SDK but I'm not completely sure.

Also any tips for speeding builds? (It took me a couple of tries to get the right version and to find the plugindir option)

```
diff --git a/org.kde.kontact.json b/org.kde.kontact.json
index b5b9166..f2c716d 100644
--- a/org.kde.kontact.json
+++ b/org.kde.kontact.json
@@ -60,6 +60,23 @@
     "rename-icon": "kontact",
     "copy-icon": true,
     "modules": [
+        {   "name": "cyrus-sasl2",
+            "config-opts": [
+               "--enable-cram",
+               "--enable-digest",
+               "--enable-otp",
+               "--enable-plain",
+                "--enable-login",
+               "--with-plugindir=/app/lib/sasl2:/usr/lib/sasl2"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "ftp://ftp.cyrusimap.org/cyrus-sasl/cyrus-sasl-2.1.26.tar.gz",
+                    "sha256": "8fbc5136512b59bb793657f36fadda6359cae3b08f01fd16b3d406f1345b7bc3"
+                }
+            ]
+        },
         {
             "name": "gpgme",
             "sources": [
```

**Comment by danvratil** [comment]
Good guess: https://github.com/flatpak/freedesktop-sdk-base/tree/1.6/meta-freedesktop/recipes-core/cyrus-sasl - do we maybe also need the patches and/or some of the configuration options they have?

**Comment by detrout** [comment]
Matching more of the Freedesktop build settings seem quite reasonable. I'll try to incorporating them on a day I'm less tired.

As for build speed, although putting the module at the top of the list was convenient for searching... maybe I should push it down further so I don't have to rebuild everything.

**Comment by danvratil** [comment]
Merged #13 and a new update build is running now, so we can close this I believe.

Thanks for your help!
